File: Data-Structure-I/118-Pascal-Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> triangle;
        for(int i=0; i<numRows; ++i){
            vector<int> col;
            for(int j=0; j<=i; ++j){
                if(j==0 || j==i) 
                    col.push_back(1);
                else
                    col.push_back(triangle[i-1][j] + triangle[i-1][j-1]);
            }
            triangle.push_back(col);
        }
        return triangle;
    }
};

File: Data-Structure-I/118-Pascal-Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = [[1]]
        for i in range(1, numRows):
            row = [1]
            for j in range(len(triangle[i-1]) - 1):
                row.append(triangle[i-1][j] + triangle[i-1][j-1])
                row.append(1)
                triangle.append(row)
        return triangle
